### ISSUE TITLE
Cannot build Ontology on RPI

### ISSUE THREAD

**Issue by ivoras** [ontio/ontology#558]
It's probably not a target platform, but worth mentioning: Ontology currently cannot be built on Raspberry Pi (and probably any other non-amd64 architecture):

```
ivoras@rpi ~/g/s/g/o/ontology> go version
go version go1.10.3 linux/arm
ivoras@rpi ~/g/s/g/o/ontology> make all
go build  -ldflags "-X github.com/ontio/ontology/common/config.Version=v1.0.1-3-g8b6fa"  -o ontology main.go
# github.com/ontio/ontology/vendor/github.com/ontio/ontology-crypto/sm2
vendor/github.com/ontio/ontology-crypto/sm2/curves.go:39:14: undefined: initP256
vendor/github.com/ontio/ontology-crypto/sm2/curves.go:40:9: undefined: p256
Makefile:20: recipe for target 'ontology' failed
make: *** [ontology] Error 2
```

For what its worth, it seems like the problem is with P256 curves, which are also implemented in https://golang.org/pkg/crypto/elliptic/ , which is cross-platform.

**Comment by AlverLyu** [comment]
The SM2 implementation is assemble accelerated for amd64, but not compatible with other platform. Will fix this soon.

**Comment by AlverLyu** [comment]
This problem has been fixed in version 1.0.2.
